pkg/resource/training_job/custom_set_update_input.go

```diff
@@ -0,0 +1,130 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Use this file if conditions need to be updated based on the latest status
+// of training job which is not evident from API response
+
+package training_job
+
+import (
+	"errors"
+
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	ackerr "github.com/aws-controllers-k8s/runtime/pkg/errors"
+	smv1alpha "github.com/aws-controllers-k8s/sagemaker-controller/apis/v1alpha1"
+	svcsdk "github.com/aws/aws-sdk-go/service/sagemaker"
+)
+
+// customSetUpdateInput modifies the input of UpdateTrainingJob.
+// Three conditions:
+// 1. Customer updates both profiler parameters: Recreate the input for profiler Rule.
+// 2. Customer only updates Profiler Config: Set the profiler rule configuration to nil to avoid validation error.
+// 3. Customer only updates Rule Configurations: Recreate the input for profiler Rule and set Profiler config to nil.
+//	  safer to do this because the "only add" behavior might reappear.
+func customSetUpdateInput(desired *resource, latest *resource, delta *ackcompare.Delta, input *svcsdk.UpdateTrainingJobInput) error {
+	if delta.DifferentAt("Spec.ProfilerConfig") && delta.DifferentAt("Spec.ProfilerRuleConfigurations") {
```

Comment:
checks have duplication? does the following work?

```
if !delta.DifferentAt("Spec.ProfilerConfig") {
  input.SetProfilerConfig(nil)
}

if !delta.DifferentAt("Spec.ProfilerRuleConfigurations") {
  input.SetProfilerRuleConfigurations(nil)
} else {
  err := handleProfilerRuleConfig(desired, latest, input)
}
```

Reply:
works